src/test/java/com/facebook/openwifirrm/optimizers/TestUtils.java

```diff
@@ -644,7 +694,77 @@ public static State createState(
 			new int[] { channelWidthA, channelWidthB },
 			new int[] { txPowerA, txPowerB },
 			new String[] { bssidA, bssidB },
-			new int[][] { clientRssisA, clientRssisB }
+			new String[][] {},
```

Comment:
`clientRssisA` and `clientRssisB` are input parameters, it's not initialization. `new String[][] { new String[clientRssis.length] }` initialize an empty 2D string Array, if I didn't declare the size, it will run into NPE

Reply:
> Ok. Above (line 624) you define the inner array size. Any particular reason for the difference?

This method is not called externally. The usage is only called by another `createState` method:
```
	public static State createState(
		int channelA,
		int channelWidthA,
		int txPowerA,
		String bssidA,
		int channelB,
		int channelWidthB,
		int txPowerB,
		String bssidB
	) {
		return createState(
			channelA,
			channelWidthA,
			txPowerA,
			bssidA,
			new int[] {},
			channelB,
			channelWidthB,
			txPowerB,
			bssidB,
			new int[] {}
		);
	}
```
You could see that actually the `clientRssisA` and `clientRssisB` are empty, so it would never try to access the first element of the empty array. I will add the size initialization as line 624 does.